tests/test_local_executor.py

```diff
@@ -518,24 +554,25 @@ def test_running_job_terminated_finalized(docker_cleanup, job_definition, tmp_wo
     assert api.get_status(job_definition).state == ExecutorState.EXECUTING
 
     status = api.terminate(job_definition)
-    assert status.state == ExecutorState.ERROR
+    job_definition.cancelled = "user"
+    assert status.state == ExecutorState.EXECUTED
     assert api.get_status(job_definition).state == ExecutorState.EXECUTED
 
-    # run.py does not run finalize for cancelled jobs
-    # status = api.finalize(job_definition)
-    # assert status.state == ExecutorState.FINALIZED
-    # assert api.get_status(job_definition).state == ExecutorState.FINALIZED
+    status = api.finalize(job_definition)
+    assert status.state == ExecutorState.FINALIZED
+    assert api.get_status(job_definition).state == ExecutorState.FINALIZED
+
+    assert job_definition.id in local.RESULTS
+    assert local.RESULTS[job_definition.id].exit_code == 137
+    assert local.RESULTS[job_definition.id].message == "Job cancelled by user"
 
     status = api.cleanup(job_definition)
     assert status.state == ExecutorState.UNKNOWN
     assert api.get_status(job_definition).state == ExecutorState.UNKNOWN
 
     assert job_definition.id not in local.RESULTS
-    # TODO: should we write these values when finalizing a cancelled job?
-    # assert local.RESULTS[job_definition.id].exit_code == 137
-    # assert local.RESULTS[job_definition.id].message == "Killed by an OpenSAFELY admin"
 
-    assert not workspace_log_file_exists(job_definition)
+    assert workspace_log_file_exists(job_definition)
```

Comment:
this is high_privace_workspace/ `{job_definition.action}.log`

https://github.com/opensafely-core/job-runner/blob/master/tests/test_local_executor.py#L102-L115